src/main/webapp/app/exercises/text/participate/text-editor.component.ts

```
@@ -131,22 +178,51 @@
+                    this.submission = participation.submissions.sort(this.submissionSortFunction).last() as TextSubmission;
+                }
+            } else {
+                this.submission = participation.submissions.sort(this.submissionSortFunction).last() as TextSubmission;
+            }
+
             setLatestSubmissionResult(this.submission, getLatestSubmissionResult(this.submission));
-            if (this.submission?.results && participation.results && (this.isAfterAssessmentDueDate || this.isAfterPublishDate)) {
+            if (
+                this.submission?.results &&
+                participation.results &&
+                (this.isAfterAssessmentDueDate || this.isAfterPublishDate || Result.isAthenaAIResult(this.submission.latestResult!))
```

Biome

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)